File: leetcode/1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        min_delete, b = 0, 0

        for c in s:
            if c == 'a':
                min_delete = min(min_delete + 1, b)
            else:
                b += 1
        
        return min_delete
            

File: leetcode/1756-minimum-deletions-to-make-string-balanced/minimum-deletions-to-make-string-balanced.cpp
class Solution {
public:
    int minimumDeletions(string s) {
        int deletions = 0;
        int bCount = 0;

        for (int i = 0; i < s.size(); i++) {
            if (s[i] == 'b') bCount++;
            else if (bCount > 0) {
                bCount--;
                deletions++;
            }
        }

        return deletions;
    }
};